src/Security/SslConnection.cs

```diff
@@ -97,8 +99,8 @@ protected virtual void Dispose (bool disposing)
 #endif
 		unsafe static SslStatus Read (IntPtr connection, IntPtr data, nint* dataLength)
 		{
-			var c = (SslConnection) GCHandle.FromIntPtr (connection).Target;
-			return c.Read (data, ref System.Runtime.CompilerServices.Unsafe.AsRef<nint> (dataLength));
+			var c = GCHandle.FromIntPtr (connection).Target as SslConnection;
+			return c!.Read (data, ref System.Runtime.CompilerServices.Unsafe.AsRef<nint> (dataLength));
```

Comment:
Does this look okay?

Reply:
Why do we need the c! rather than doing an as, do an is, correct? Then you get the ! out.

Reply:
I think and invalid cast exception would be a good option https://docs.microsoft.com/en-us/dotnet/api/system.invalidcastexception?view=net-6.0 but we should ping @rolfbjarne

Reply:
This is called from native code, so any exception is probably going to have rather disastrous consequences.

On the other hand, if this code fails, something disastrous already happened, so surfacing that disaster might be the best thing to do.

My vote goes for not changing the behavior (when in doubt, keep doing the same thing is usually the safest choice), so something like this:

```cs
			var c = (SslConnection) GCHandle.FromIntPtr (connection).Target!;
 			return c.Read (data, ref System.Runtime.CompilerServices.Unsafe.AsRef<nint> (dataLength));
```